pkg/podrestartmonitor/pod_restart_monitor.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package podrestartmonitor
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"time"
+
+	"github.com/golang/glog"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	"k8s.io/node-problem-detector/pkg/problemdaemon"
+	"k8s.io/node-problem-detector/pkg/types"
+)
+
+const PodRestartMonitorName = "pod-restart-monitor"
+
+func init() {
+	problemdaemon.Register(
+		PodRestartMonitorName,
+		types.ProblemDaemonHandler{
+			CreateProblemDaemonOrDie: NewMonitorOrDie,
+			CmdOptionDescription:     "Set to config file paths."})
+}
+
+type podRestartMonitor struct {
+	configPath string
+	config     MonitorConfig
+	stop       chan struct{}
+	clientset  *kubernetes.Clientset
+	statuses   chan *types.Status
+}
+
+// NewMonitorOrDie create a new podRestartMonitor, panic if error occurs.
+func NewMonitorOrDie(configPath string) types.Monitor {
+	l := &podRestartMonitor{
+		configPath: configPath,
+	}
+
+	f, err := ioutil.ReadFile(configPath)
+	if err != nil {
+		glog.Fatalf("Failed to read configuration file %q: %v", configPath, err)
+	}
+	if l.config, err = parseConfig(f); err != nil {
+		glog.Fatalf("Failed to parse configuration file %q: %v", configPath, err)
+	}
+
+	glog.Infof("Finish parsing pod restart monitor config file %s: %+v", l.configPath, l.config)
+	l.stop = make(chan struct{})
+
+	return l
+}
+
+func parseConfig(f []byte) (config MonitorConfig, err error) {
+	if err = json.Unmarshal(f, &config); err != nil {
+		return config, err
+	}
+
+	// Apply default configurations
+	if config.CheckInterval == 0 {
+		config.CheckInterval = 5 * time.Minute
+	}
+	if config.RestartThreshold == 0 {
+		config.RestartThreshold = 5
+	}
+	if config.Namespace == "" {
+		config.Namespace = "kube-system"
+	}
+	if config.PodSelector == "" {
+		return config, fmt.Errorf("did not specify a PodSelector")
+	}
```

Comment:
Although it would make the process of filtering pods more complex, it sure would be nice to be able to filter by annotations or by priority classes, in addition to labels. Feel free to ignore my comment here if you think it'd be a pain to implement, though.